includes/models/class-wc-stripe-subscription.php

```diff
@@ -0,0 +1,298 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class WC_Stripe_Subscription
+ *
+ * Wrapper for the original WC_Subscription class to allow custom getters and setter with the extension's specific metadata.
+ */
+class WC_Stripe_Subscription {
+	/**
+	 * Meta key for the Stripe source ID.
+	 *
+	 * @var string
+	 */
+	const META_STRIPE_SOURCE_ID = '_stripe_source_id';
+
+	/**
+	 * Meta key for the customer ID.
+	 *
+	 * @var string
+	 */
+	const META_STRIPE_CUSTOMER_ID = '_stripe_customer_id';
+
+	/**
+	 * Meta key for the card ID.
+	 *
+	 * @var string
+	 */
+	const META_STRIPE_CARD_ID = '_stripe_card_id';
+
+	/**
+	 * Meta key for the delayed update payment method.
+	 *
+	 * @var string
+	 */
+	const META_DELAYED_UPDATE_PAYMENT_METHOD_ALL = '_delayed_update_payment_method_all';
+
+	/**
+	 * Meta key for the Stripe refund ID.
+	 *
+	 * @var string
+	 */
+	const META_STRIPE_REFUND_ID = '_stripe_refund_id';
+
+	/**
+	 * The subscription object.
+	 *
+	 * @var WC_Subscription
+	 */
+	private $wc_subscription;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param $subscription WC_Subscription|WC_Stripe_Subscription The subscription object.
+	 */
+	public function __construct( $subscription ) {
+		if ( $subscription instanceof WC_Stripe_Subscription ) {
+			$this->wc_subscription = $subscription->wc_subscription;
+		} elseif ( class_exists( 'WC_Subscription' ) && $subscription instanceof WC_Subscription ) {
+			$this->wc_subscription = $subscription;
+		}
+
+		if ( ! $this->wc_subscription ) {
+			$type = is_object( $subscription ) ? get_class( $subscription ) : gettype( $subscription );
+			throw new InvalidArgumentException( 'WC_Stripe_Subscription requires a valid underlying WC_Subscription; supplied $subscription argument was ' . $type );
+		}
+	}
+
+	/**
+	 * Magic method to call methods on the underlying WC_Subscription object.
+	 *
+	 * @param string $name The name of the method to call.
+	 * @param array $arguments The arguments to pass to the method.
+	 */
+	public function __call( $name, $arguments ) {
+		if ( method_exists( 'WC_Subscription', $name ) ) {
+			return call_user_func_array( [ $this->wc_subscription, $name ], $arguments );
+		}
+
+		throw new BadMethodCallException( 'Method ' . $name . ' does not exist in WC_Stripe_Subscription' );
+	}
+
+	/**
+	 * Wrapper to return an order using the extension's custom WC_Stripe_Subscription class.
+	 *
+	 * @param $subscription_id int Subscription ID.
+	 * @return bool|WC_Stripe_Subscription
+	 */
+	public static function get_by_id( $subscription_id ) {
+		$subscription = function_exists( 'wcs_get_subscription' ) ? wcs_get_subscription( $subscription_id ) : false;
+		return $subscription ? self::to_instance( $subscription ) : false;
```

Comment:
Super-minor nit; definitely not blocking: these two lines include a lot of logic behind the two ternary conditions. It might be clearer and simpler if we expanded the code out a bit so the exit conditions are super-easy to parse.
```suggestion
		if ( ! function_exists( 'wcs_get_subscription' ) ) {
			return false;
		}
		$subscription = wcs_get_subscription( $subscription_id );
		if ( ! $subscription ) {
			return false;
		}
		return self::to_instance( $subscription );
```